examples/keras_cv/data_augmentation_with_keras_cv.py

```diff
@@ -0,0 +1,388 @@
+in a way that is not possible with a sparse label representation.
+"""
+
+IMAGE_SIZE = (224, 224)
+BATCH_SIZE = 32
+num_classes = ds_info.features["label"].num_classes
+
+
+def prepare(image, label):
+    image = tf.image.resize(image, IMAGE_SIZE)
+    image = tf.cast(image, tf.float32)
+    label = tf.one_hot(label, num_classes)
+    return {"images": image, "labels": label}
+
+
+def prepare_dataset(ds, split):
+    if split == "train":
+        return (
+            ds.map(lambda x, y: prepare(x, y), num_parallel_calls=AUTOTUNE)
+            .shuffle(10 * BATCH_SIZE)
+            .batch(BATCH_SIZE)
+        )
+    if split == "test":
+        return ds.map(lambda x, y: prepare(x, y), num_parallel_calls=AUTOTUNE).batch(
+            BATCH_SIZE
+        )
+
+
+def load_dataset(split="train"):
+    ds = data[split]
+    return prepare_dataset(ds, split)
+
+
+train_ds = load_dataset()
+
+"""
+Let's inspect some samples from our dataset:
+"""
+
+
+def visualize_dataset(ds, title):
+    plt.figure(figsize=(6, 6)).suptitle(title, fontsize=18)
+    for i, samples in enumerate(iter(ds.take(9))):
+        images = samples["images"]
+        plt.subplot(3, 3, i + 1)
+        plt.imshow(images[0].numpy().astype("uint8"))
+        plt.axis("off")
+    plt.show()
+
+
+visualize_dataset(train_ds, title="Before Augmentation")
+
+"""
+Great!  Now we can move onto the augmentation step.
+"""
+
+"""
+## RandAugment: Your One Stop Shop for Data Augmentation
+"""
+
+"""
+While `CutMix` and `MixUp` have produced some great interclass examples, we still need to
+perturb the images.  This prevents the model from overfitting the training distribution,
+and improves the likelihood that the model can generalize to out of distribution
+examples.
+
+To remedy this issue KerasCV offers RandAugment, a general purpose image data
+augmentation algorithm.  RandAugment has been shown to provide improved image
+classification results across numerous datasets, and only has two hyperparameters to
+configure.
+
+Let's check out KerasCV's RandAugment implementation.
+"""
+
+rand_augment = preprocessing.RandAugment(
+    value_range=(0, 255),
+    augmentations_per_image=3,
+    magnitude=0.3,
+    magnitude_stddev=0.2,
+    rate=0.5,
+)
+
+"""
+To use `RandAugment` you need to provide a few values:
+
+- `value_range` describes the range of values covered in your images
+- `magnitude` is a value between 0 and 10, describing the strength of the perturbations
+applied
+- `num_layers` is an integer telling the layer how many augmentations to apply to each
+individual image
+- (Optional) `magnitude_standard_deviation` allows `magnitude` to be randomly sampled
+from a distribution with a standard deviation of `magnitude_standard_deviation`
+- (Optional) `probability_to_apply` indicates the probability to apply the augmentation
+applied at each layer.
+
+That's all!  Let's apply our RandAugment layer:
+"""
+
+
+@tf.function
+def apply_rand_augment(inputs):
+    result = inputs.copy()
```

Comment:
Is this line really necessary?

Reply:
Unfortunately it is and I'm not really sure why.  It's really strange to me; but I get an error saying that tf functions may not modify their inputs.  The weirder thing, this doesn't happen when I tf.function-ify in my classes' call methods.

Reply:
Definitely something we should clear up, since it's likely to be an error that users will commonly run into, and they may not figure out the workaround on their own.